packages/wdio-config/src/lib/ConfigParser.ts

```diff
@@ -297,6 +310,13 @@ export default class ConfigParser {
         // Remove any duplicate tests from the final specs array
         specs = [...new Set(specs)]
 
+        // For Cucumber, filter the specs according to the tag expression
+        // Some workers would only spawn to then skip the spec (Feature) file
+        // Filtering at this stage can prevent the spawning of a massive number of workers
+        if (this._config.framework === 'cucumber' && this._config.cucumberOpts?.tagExpression) {
+            specs = this.filterSpecsByTagExpression(specs)
+        }
```

Comment:
This should be done in the `@wdio/cucumber-framework` package

Reply:
See https://github.com/webdriverio/webdriverio/pull/10330#discussion_r1187717431